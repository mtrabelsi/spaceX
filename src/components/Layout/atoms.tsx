import s from 'styled-components';
import { defaultColor, tabletWidth } from '../Global'

export const LayoutMain = s.main`
    padding: 20px;
    //ALWAYS centred
    max-width: ${tabletWidth};
    margin: 0 auto;
    color: ${defaultColor}
`;