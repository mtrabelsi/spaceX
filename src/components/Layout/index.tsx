import React from 'react';
import { LayoutMain } from './atoms';
import { RouteComponentProps } from 'react-router-dom';
import { Button } from '../Button/atoms';

type LayoutProps = {
    style?: object,
    showBackButton?: boolean
}

const Layout: React.FC<LayoutProps | RouteComponentProps> = (props) => {
    const { showBackButton, style, history } = props as (LayoutProps & RouteComponentProps)

    return(<LayoutMain style={style}>
        {showBackButton && <section>
            <Button onClick={e => history.goBack()}>
                Home
            </Button>
        </section>}
        {props.children}
    </LayoutMain>)
}


export default Layout;