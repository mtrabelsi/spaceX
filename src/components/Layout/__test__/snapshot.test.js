import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '..';

it('Should consider all props and generate unique html based on that', () => {
  const tree = renderer
    .create(
      <Layout
        style={{ width: 500 }}
        title="sample title"
        topBarStyle="100px"
        showBackButton
        history={{ goBack: () => {} }}
      >
        <span>some child data</span>
      </Layout>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
