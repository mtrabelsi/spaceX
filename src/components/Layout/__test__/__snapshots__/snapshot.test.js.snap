// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should consider all props and generate unique html based on that 1`] = `
<section
  className="sc-AxjAm bMhmYx"
  style={
    Object {
      "width": 500,
    }
  }
>
  <section
    className="sc-AxiKw keaLxW"
    style={
      Object {
        "0": "1",
        "1": "0",
        "2": "0",
        "3": "p",
        "4": "x",
        "flexWrap": "wrap",
      }
    }
  >
    <button
      className="sc-AxhCb kYAQok"
      onClick={[Function]}
    >
      <img
        className="sc-Axmtr kdbydJ"
        src="Left-Arrow.png"
      />
      <span
        className="sc-AxhUy jSdugU"
      >
        Home
      </span>
    </button>
    <h1
      className="sc-AxirZ bEWkiU"
    >
      sample title
    </h1>
  </section>
  <span>
    some child data
  </span>
</section>
`;
